### PR TITLE
Use chokidar

### LINKED ISSUE #1197: A high severity vulnerability introduced in oc
Hi, a vulnerability https://snyk.io/vuln/SNYK-JS-MERGE-1040469 is introduced in oc via:
●  oc@0.48.19 ➔ watch@1.0.2 ➔ exec-sh@0.2.2 ➔ merge@1.2.1

However, **_watch_** is a legacy package, which has not been maintained for about 4 years.
Is it possible to migrate **_watch_** to other package or remove it to remediate this vulnerability?

I noticed a migration record in other js repo for **_watch_**:

● in @google/clasp, version 2.3.2 ➔ 2.4.0, Migrate from watch to chokidar via [commit](https://github.com/google/clasp/commit/003d422b5b5fdae4ece6eb6c24db2591b53fc956)
● in forever-monitor, version 1.5.2 ➔ 1.6.0, Migrate from watch to chokidar via [commit](https://github.com/foreversd/forever-monitor/commit/12e35fcd17f0e7beace9a62dfd949300fe4fb371)

Are there any efforts planned that would remediate this vulnerability or migrate **_watch_**?

Thanks
; )

### PR DESCRIPTION
**Pull request by ricardo-devis-agullo**
Closes #1197 

Use chokidar in favor of watch, since the latter is unmantained. I disabled `polling` (although it's disabled by default) since for our use case I don't think it's needed and should be faster ([check here](https://github.com/paulmillr/chokidar#performance)). This setup worked the same in local with a couple of components, so should be fine.

I also took the opportunity to upgrade devDependencies

### CODE REVIEW

File: src/cli/domain/watch.js [hunk 1]
```diff
@@ -1,33 +1,29 @@
...
-      },
-      (fileName, currentStat, previousStat) => {
-        if (!!currentStat || !!previousStat) {
-          const componentDir = dirs.find(dir =>
-            Boolean(fileName.match(escapeRegularExpression(dir + path.sep)))
-          );
-          changed(null, fileName, componentDir);
-        }
-      }
+  const watcher = chokidar.watch(path.resolve(baseDir), {
```
Comment (pbazydlo) [inline src/cli/domain/watch.js:9]: is it possible for this code to throw? or was it just expected syntax before with `watch`?

**Comment by pbazydlo** [comment]
@ricardo-devis-agullo have you been able to test it on both mac and windows?

File: src/cli/domain/watch.js (hunk 1 above)
Comment (ricardo-devis-agullo) [reply src/cli/domain/watch.js:9]: No, chokidar uses event emitter, so errors will go to the error event.

**Comment by ricardo-devis-agullo** [comment]
No, but I can test on Mac

**Comment by ricardo-devis-agullo** [comment]
Tested on mac and it works

**Review by pbazydlo — APPROVED** [review]

**Pull request merged**